src/tag.rs

```diff
@@ -0,0 +1,723 @@
+    pub fn name_matches<T: Into<Option<S>>, S: Into<String>>(mut self, name: T) -> Self {
+        self.name_matches = name.into().map(Into::into);
+        self
+    }
+
+    /// Search by [`Tag::name`], exactly.
+    ///
+    /// This is a convenience function for setting a single value. For multiple values, see
+    /// [`Self::names`].
+    pub fn name<T: Into<Option<S>>, S: Into<String>>(mut self, name: T) -> Self {
+        self.name = name.into().into_iter().map(Into::into).collect();
+        self
+    }
+
+    /// Search by [`Tag::name`], exactly.
+    ///
+    /// For a convenience function to set a single value, see [`Self::name`].
+    pub fn names<I: IntoIterator<Item = S>, S: Into<String>>(mut self, names: I) -> Self {
+        self.name = names.into_iter().map(Into::into).collect();
+        self
+    }
+
+    /// Search by [`Tag::category`].
+    ///
+    /// This is a convenience function for setting a single value. For multiple values, see
+    /// [`Self::categories`].
+    pub fn category<T: Into<Option<Category>>>(mut self, category: T) -> Self {
+        self.category = category.into().into_iter().collect();
+        self
+    }
+
+    /// Search by [`Tag::category`], exactly.
+    ///
+    /// For a convenience function to set a single value, see [`Self::category`].
+    pub fn categories<I: IntoIterator<Item = Category>>(mut self, categories: I) -> Self {
+        self.category = categories.into_iter().collect();
+        self
+    }
+
+    /// Whether to hide results that have no posts.
+    pub fn hide_empty(mut self, hide_empty: bool) -> Self {
+        self.hide_empty = hide_empty;
+        self
+    }
+
+    /// Whether to only show results that have an associated wiki page.
+    pub fn has_wiki(mut self, has_wiki: bool) -> Self {
+        self.has_wiki = has_wiki;
+        self
+    }
+
+    /// Whether to only show results that either are artists or have a matching artist tag (eg.
+    /// `lips` and `lips_(artist)`).
+    pub fn has_artist(mut self, has_artist: bool) -> Self {
+        self.has_artist = has_artist;
+        self
+    }
+}
+
+impl Client {
+    /// Returns a Stream over all the [`Tag`]s matching the search query.
+    ///
+    /// ```no_run
+    /// # use rs621::client::Client;
+    /// # use rs621::tag::Query;
+    /// use futures::prelude::*;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() -> rs621::error::Result<()> {
+    /// let client = Client::new("https://e926.net", "MyProject/1.0 (by username on e621)")?;
+    ///
+    /// let query = Query::new()
+    ///     .name("fluffy")
+    ///     .per_page(1);
+    ///
+    /// let tag_stream = client.tag_search(query);
+    /// futures::pin_mut!(tag_stream);
+    ///
+    /// while let Some(tag) = tag_stream.next().await {
+    ///     assert_eq!(tag?.name, "fluffy");
+    /// }
+    /// # Ok(()) }
+    /// ```
+    #[cfg(not(any(target_arch = "wasm32", target_arch = "wasm64")))]
```

Comment:
```suggestion
    #[cfg(not(target_family = "wasm"))]
```

same for the other ones